lib/node/httpStack.js

```diff
@@ -47,10 +69,17 @@ class Request {
     return new Promise((resolve, reject) => {
       let options = parse(this._url);
       options.method = this._method;
-      options.headers = this._headers;
+      options.headers = {
+        ...(this._requestOptions.headers || {}),
+        ...this._headers
+      };
       if (body && body.size) options.headers["Content-Length"] = body.size;
+      options = {
+        ...this._requestOptions,
+        ...options
+      };
 
-      let req = this._request = (options.protocol !== "https:") ? http.request(options) : https.request(options);
+      let req = (this._request = options.protocol !== "https:" ? http.request(options) : https.request(options));
```

Comment:
Can you explain why the change of these parenthesis is about?

Reply:
I have no clue what that was :)) reverted back